source/Calamari.Aws/Kubernetes/Discovery/AwsAuthenticationDetails.cs

```diff
@@ -9,102 +9,64 @@ namespace Calamari.Aws.Kubernetes.Discovery
     public class AwsAuthenticationDetails : ITargetDiscoveryAuthenticationDetails
     {
         const string DefaultSessionName = "OctopusKubernetesClusterDiscovery";
-        
+
         /// <exception cref="AggregateException">
         /// If both InstanceProfile and EnvironmentVariable Credentials fail.
         /// Contains AmazonClientExceptions for both InstanceProfile and EnvironmentVariable failures</exception>
         /// <exception cref="AmazonClientException">If Basic (Account) Credentials fail</exception>
-        public bool TryGetCredentials(ILog log, out AWSCredentials credentials)
+        public AWSCredentials GetCredentials()
         {
-            credentials = null;
-            if (Credentials.Type == "account")
-            {
-                try
-                {
-                    credentials = new BasicAWSCredentials(Credentials.Account.AccessKey, Credentials.Account.SecretKey);
-                }
-                // Catching a generic Exception because AWS SDK throws undocumented exceptions.
-                catch (Exception e)
-                {
-                    log.Warn("Unable to authorise credentials, see verbose log for details.");
-                    log.Verbose($"Unable to authorise credentials for Account: {e}");
-                    return false;
-                }
-            }
-            else
-            {
-                try
-                {
-                    // If not currently running on an EC2 instance,
-                    // this will throw an exception.
-                    credentials = new InstanceProfileAWSCredentials();
-                }
-                // Catching a generic Exception because AWS SDK throws undocumented exceptions.
-                catch (Exception instanceProfileException)
-                {
-                    try
-                    {
-                        // The last attempt is trying to use Environment Variables.
-                        credentials = new EnvironmentVariablesAWSCredentials();
-                    }
-                    // Catching a generic Exception because AWS SDK throws undocumented exceptions.
-                    catch (Exception environmentVariablesException)
-                    {
-                        log.Warn("Unable to authorise credentials, see verbose log for details.");
-                        log.Verbose($"Unable to authorise credentials for Instance Profile: {instanceProfileException}");
-                        log.Verbose($"Unable to authorise credentials for Environment Variables: {environmentVariablesException}");
-                        return false;
-                    }
-                }
-            }
+            var credentials = Credentials.Type == "account"
+                ? new BasicAWSCredentials(Credentials.Account.AccessKey, Credentials.Account.SecretKey)
+                : FallbackCredentialsFactory.GetCredentials();
```

Comment:
Using FallbackCredentialsFactory from AWS library to go through all fallback credential options in default order.